README.md

```diff
@@ -79,6 +80,56 @@ Query tags are visible in Snowflake's query history and can be used for:
 - Debugging performance issues
 - Auditing database access
 
+## Error Logging and Metadata
+
+Snowflex automatically enriches error logs with contextual metadata to help with debugging and monitoring. When queries fail, detailed information about the error context is added to Logger metadata.
+
+### Available Metadata Keys
+
+The following metadata keys are set when errors occur:
+
+- `:snowflex_account_name` - Your Snowflake account identifier
+- `:snowflex_username` - The username used for the connection
+- `:snowflex_query_id` - Snowflake's unique query identifier
+- `:snowflex_statement` - The SQL statement that failed
+- `:snowflex_warehouse` - The default warehouse used for the query (if configured)
+- `:snowflex_role` - The default role used for the query (if configured)
+- `:snowflex_database` - The default database used for the query (if configured)
+- `:snowflex_schema` - The default schema used for the query (if configured)
+
+### Configuring Logger
+
+To capture this metadata in your logs, configure Logger in your `config/config.exs` and include any keys you are interested to seeing in the metadata:
+
+```elixir
+config :logger, :console,
+  metadata: [
+    :snowflex_account_name,
+    :snowflex_username,
+    :snowflex_warehouse,
+    :snowflex_role,
+    :snowflex_database,
+    :snowflex_schema,
+    :snowflex_query_id,
+    :snowflex_statement
+  ]
+```
```

Comment:
I'd probably just drop a line like `See [Logger.metadata/1] for more information`

This will ensure we don't have to keep our README in sync with Elixir's logger.

Reply:
agreed